### PR TITLE
Refactor integration methods to use only survival-based approach (res…

### PR DESCRIPTION
**Pull request by Mdraugelis**
…olves #44)

Core Changes:
- Remove 'average' and 'weighted_recent' integration methods from risk_integration.py
- Simplify integrate_window_risk() function to only use survival-based integration
- Update ml_simulation.py temporal functions to remove integration_method parameter
- Clean up type annotations and remove unused Literal import

Updated Functions:
- integrate_window_risk(): Now only accepts timestep_duration parameter
- generate_temporal_predictions(): Removed integration_method parameter
- generate_temporal_ml_predictions(): Removed integration_method parameter
- benchmark_temporal_ml_performance(): Configs no longer need integration_method

Test Updates:
- Refactored test_risk_integration.py to focus on survival integration properties
- Updated test_temporal_ml_simulation.py to remove multi-method comparisons
- All integration tests now validate only survival-based approach
- Maintained test coverage for edge cases and validation

Benefits:
- Simplified API with single, mathematically sound integration method
- Reduced code complexity and maintenance burden
- Survival integration properly accounts for cumulative risk over time
- Cleaner type annotations and function signatures

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/pop_ml_simulator/ml_simulation.py [hunk 1]
```diff
@@ -443,9 +437,9 @@ def generate_temporal_predictions(
         # Calculate temporal correlation metrics
         n_patients, n_timesteps = temporal_risk_matrix.shape
         integration_info = {
-            'integration_method': str(integration_method),
-            'window_start': float(prediction_start_time),
-            'window_length': float(prediction_window_length),
+            'integration_method': 'survival',
```
[the remark this answers is not in the document]
Comment (Mdraugelis) [reply src/pop_ml_simulator/ml_simulation.py:440]: not worried about this.
